realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.kt

```diff
@@ -1921,9 +1921,85 @@ class RealmProxyClassGenerator(private val processingEnvironment: ProcessingEnvi
                         endControlFlow()
                         emitEmptyLine()
                     }
-                    Utils.isRealmSet(field) -> {
-                        // TODO: Set
-                        emitSingleLineComment("TODO: Set")
+                    Utils.isRealmModelSet(field) -> {
+                        val genericType: TypeMirror = Utils.getGenericType(field)!!
+                        val isEmbedded = Utils.isFieldTypeEmbedded(genericType, classCollection)
+
+                        emitEmptyLine()
+                        emitStatement("OsSet %1\$sOsSet = new OsSet(table.getUncheckedRow(objKey), columnInfo.%1\$sColKey)", fieldName)
+                        emitStatement("RealmSet<%s> %sSet = ((%s) object).%s()", genericType, fieldName, interfaceName, getter)
+                        if (isEmbedded) {
+                            // throw not supported
```

Comment:
Not sure I understand this comment. Also, I cannot remember the full implementation details, but should we not throw and cancel compilation if the type is an embedded object? Or do we handle that case in runtime?

Reply:
Forgot to add the piece of code that handles the case.